api/envoy/config/core/v3/socket_option.proto

```diff
@@ -49,6 +49,9 @@ message SocketOption {
 
     // Socket options are applied after calling listen()
     STATE_LISTENING = 2;
+
+    // Socket options are applied in any state.
+    STATE_ANY = 3;
```

Comment:
I'm not sure I like adding this here, because it's in the public API, and setting this would cause an option to be set multiple times in various socket states, which is almost certainly not what we want.

Reply:
I removed this